emacs_erlang_yaemep_support.erl

```diff
@@ -92,6 +94,206 @@ erlang_project_dir(ErlangFilePath) ->
         Path -> Path
     end.
 
+%% The `find_files' is a like `filelib:wildcard', but takes care
+%% to avoid visiting paths in symlink loops more than once.
+%% If the project is large and symlinks point many levels root-wards,
+%% filelib:wildcard can take very long to complete, and can consume large
+%% amounts of memory as the same files are found multiple times.
+%%
+%% Support patterns on these forms:
+%% - TopDir/**/*.{erl,hrl,c,h}
+%% - TopDir/**/*.beam
+%% - TopDir/**/xyz.erl
+%% - TopDir/*.beam
+%% - TopDir/somefile.ext
+%% If TopDir is missing, use current directory, but don't include it
+%% in the results
+-spec find_files(Pattern::string()) -> [FileName::string()].
+find_files(Wildcard) ->
+    {TopDir, Pattern, RT} = analyze_wildcard(Wildcard),
+    Visited0 = new_set(), % containing paths for files, dirs and symlinks
+    Acc0 = [],            % matching files
+    {_Visited, Acc} = traverse_fs(TopDir, Pattern, RT, Visited0, Acc0),
+    lists:reverse(Acc).
```

Comment:
I think it should be easy to detect if `find_files` is given a pattern that it cannot handle and make it fallback on `filelib:wildcard` in that case (e.g., by making use of a regular expression). That will, of course, not solve the issue with symlink loops when such patterns are given, but I think it is worth the effort since such patterns should be rare and YAEMEP will then still be able to handle such patterns when symlink loops do not exist.

Reply:
Done. (Though not with a regexp... Don't know exactly how such a regexp would look like)